### PR TITLE
add ability to define custom grok patterns within processor config

### PR DESCRIPTION
**Pull request by talevy**
adds ability to do something like this:

``` json
{
  "grok" : {
    ...
    "pattern_definitions" : {
      "MY_PATTERN" : "hello!"
    }
  }
}
```

### CODE REVIEW

File: docs/plugins/ingest.asciidoc [hunk 1]
```diff
@@ -295,6 +296,28 @@ This pipeline will insert these named captures as new fields within the document
...
+
+[source,js]
+--------------------------------------------------
+{
+  "description" : "...",
+  "processors": [
+    {
+      "grok": {
+        "match_field": "message",
+        "match_pattern": "%{IP:client} %{WORD:method}"
```
Comment (martijnvg) [inline docs/plugins/ingest.asciidoc]: maybe also use the custom definitions in the `match_pattern`?

**Comment by martijnvg** [comment]
One small docs comment OTT LTGM

File: docs/plugins/ingest.asciidoc (hunk 1 above)
Comment (talevy) [inline docs/plugins/ingest.asciidoc]: oo, meant to. missed it. thanks!

**Comment by talevy** [comment]
updated doc

**Pull request merged**